frontend/src/components/Review/Rating.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { ReactElement } from 'react';
+import styles from './Rating.module.scss';
+import FavoriteIcon from '@material-ui/icons/Favorite';
+import FavoriteBorderIcon from '@material-ui/icons/FavoriteBorder';
+
+type Props = {
+  readonly rating: number;
+};
+const filled_hearts = (n: number) => {
+  let rows = [];
+  for (let i = 0; i < n; i++) {
+    rows.push(<FavoriteIcon fontSize="small" className={styles.rating} />);
```

Comment:
For both this and the `empty_hearts` function, you can condense the entire function body to just be `[...Array(n)].map(() =><FavoriteIcon fontSize="small" className={styles.rating} />)`. 

This basically creates an empty array of `n` length and maps over it creating a new copy with `FavoriteIcon` components contained in each element.